tests/js/client/shell/shell-key-generation-one-shard-cluster.js

```diff
@@ -0,0 +1,317 @@
+/*jshint globalstrict:false, strict:false */
+/*global arango, assertEqual, assertTrue, assertFalse, assertMatch */
+
+// //////////////////////////////////////////////////////////////////////////////
+// / DISCLAIMER
+// /
+// / Copyright 2014-2025 ArangoDB GmbH, Cologne, Germany
+// / Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+// /
+// / Licensed under the Business Source License 1.1 (the "License");
+// / you may not use this file except in compliance with the License.
+// / You may obtain a copy of the License at
+// /
+// /     https://github.com/arangodb/arangodb/blob/devel/LICENSE
+// /
+// / Unless required by applicable law or agreed to in writing, software
+// / distributed under the License is distributed on an "AS IS" BASIS,
+// / WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// / See the License for the specific language governing permissions and
+// / limitations under the License.
+// /
+// / Copyright holder is ArangoDB GmbH, Cologne, Germany
+// /
+/// @author Test Author Max Neunhoeffer
+/// @author Copyright 2025, ArangoDB GmbH, Cologne, Germany
+// //////////////////////////////////////////////////////////////////////////////
+
+const jsunity = require("jsunity");
+const arangodb = require("@arangodb");
+const db = arangodb.db;
+const internal = require("internal");
+const isCluster = internal.isCluster();
+let { instanceRole } = require('@arangodb/testutils/instance');
+let IM = global.instanceManager;
+
+const cn = "UnitTestsKeyGenerationOneShard";
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test suite: key generation in one-shard collection
+////////////////////////////////////////////////////////////////////////////////
+
+function KeyGenerationOneShardSuite() {
+  'use strict';
+
+  return {
+
+    setUp: function () {
+      // Clean up any existing collection
+      try {
+        db._drop(cn);
+      } catch (err) {
+        // Collection might not exist, ignore
+      }
+    },
+
+    tearDown: function () {
+      IM.debugClearFailAt('always-fetch-new-cluster-wide-uniqid', instanceRole.dbServer);
+      // Clean up the test collection
+      try {
+        db._drop(cn);
+      } catch (err) {
+        // Collection might not exist, ignore
+      }
+    },
+
+    testKeyGenerationOneShardCollection: function () {
+      // Create a collection with exactly one shard
+      let c = db._create(cn, { 
+        numberOfShards: 1,
+        keyOptions: { 
+          type: "traditional",
+          allowUserKeys: false 
+        }
+      });
+
+      // Verify the collection has exactly one shard
+      assertEqual(1, c.shards().length, "Collection should have exactly one shard");
+      assertEqual(cn, c.name(), "Collection name should match");
+
+      // Set failure points to always fetch new cluster-wide unique IDs from agency on dbservers:
+      IM.debugSetFailAt('always-fetch-new-cluster-wide-uniqid', instanceRole.dbServer);
+
```

Comment:
Test it self looks good, however I am wondering if we should create (and drop) one document before the uniqId adjustment, to validate that _key is below the minimum range?